go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -128,8 +128,14 @@ func (npw *nodePortWatcher) AddService(service *kapi.Service) {
 	addSharedGatewayIptRules(service, npw.nodeIP)
 }
 
-func (npw *nodePortWatcher) UpdateService(new, old *kapi.Service) {
```

Comment:
According to the commit: the `Update was accidentally trying to update from "new" to "old" `, this is not really true right?

Reply:
why not? the old, new were inversed so the npw.DeleteService would try to delete the new one, and AddService would try to add the old one lol

Reply:
Not sure what you mean, `npw.DeleteService` was deleting the old and `AddService` was adding the new:

https://github.com/ovn-org/ovn-kubernetes/blob/3ad8b9c9e3de74d0add7b9c223878ab30cd4cd86/go-controller/pkg/node/gateway_shared_intf.go#L131-L137

Anyways, it's not that important.

Reply:
yes but @dave-tucker abstracted this, so that this function was being called by another UpdateService function, which had the parameters reversed in the function definition and call